core/src/main/java/com/linecorp/armeria/client/HttpClientDelegate.java

```diff
@@ -65,6 +64,12 @@ public HttpResponse execute(ClientRequestContext ctx, HttpRequest req) throws Ex
         if (throwable != null) {
             return earlyFailedResponse(throwable, ctx, req);
         }
+        if (req != ctx.request()) {
+            return earlyFailedResponse(
+                    new IllegalStateException("ctx.request() does not match the actual request; " +
+                                              "did you forget to call ctx.updateRequest() in your decorator?"),
+                    ctx, req);
+        }
```

Comment:
By making sure `req == ctx.request()`, we don't need to validate `req.path` anymore because `ctx.updateRequest()` or `ctx.newDerivedContext()` validated it already.